aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/servlet/AwsAlbHttpServletRequest.java

```diff
@@ -0,0 +1,457 @@
+import com.amazonaws.serverless.proxy.model.ContainerConfig;
+import com.amazonaws.serverless.proxy.model.Headers;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.ApplicationLoadBalancerRequestEvent;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.SecurityContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.security.Principal;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AwsAlbHttpServletRequest extends AwsHttpServletRequest{
+
+    //-------------------------------------------------------------
+    // Variables - Private
+    //-------------------------------------------------------------
+
+    private ApplicationLoadBalancerRequestEvent request;
+    private SecurityContext securityContext;
+    private AwsAsyncContext asyncContext;
+    private static Logger log = LoggerFactory.getLogger(AwsProxyHttpServletRequest.class);
+    private ContainerConfig config;
+
+    public AwsAlbHttpServletRequest(ApplicationLoadBalancerRequestEvent albRequest, Context lambdaContext, SecurityContext awsSecurityContext) {
+        this(albRequest, lambdaContext, awsSecurityContext, LambdaContainerHandler.getContainerConfig());
+    }
+
+    public AwsAlbHttpServletRequest(ApplicationLoadBalancerRequestEvent albRequest, Context lambdaContext, SecurityContext awsSecurityContext, ContainerConfig config) {
+        super(lambdaContext);
+        this.request = albRequest;
+        this.securityContext = awsSecurityContext;
+        this.config = config;
+    }
+
+    public ApplicationLoadBalancerRequestEvent getAwsProxyRequest() {
+        return this.request;
+    }
+    @Override
+    public String getAuthType() {
+        return securityContext.getAuthenticationScheme();
+    }
+
+
+    @Override
+    public Cookie[] getCookies() {
+        return AwsHttpServletRequestHelper.getCookies(request.getMultiValueHeaders());
+    }
+
+
+    @Override
+    public long getDateHeader(String s) {
+       return AwsHttpServletRequestHelper.getDateHeader(request.getMultiValueHeaders(), s, log);
+    }
+
+
+    @Override
+    public String getHeader(String s) {
+        return AwsHttpServletRequestHelper.getHeader(
+                request.getMultiValueHeaders(),
+                null,
+                null,
+                s,
+                getHeaderValues(s)
+        );
+    }
+
+
+    @Override
+    public Enumeration<String> getHeaders(String s) {
+        return AwsHttpServletRequestHelper.getHeaders(request.getMultiValueHeaders(), s);
```

Comment:
Both `AwsAlbHttpServletRequest` and `AwsProxyHttpServletRequest` extend `AwsHttpServletRequest`. Instead of overriding all methods twice in subclasses and delegating to the helper, this should be done in `AwsHttpServletRequest` directly.

Reply:
`AwsHttpServletRequest` doesn't have access to the underlying request objects. I can't think of a way to move the methods there if I can't access the request objects (`APIGatewayProxyRequestEvent` and `ApplicationLoadBalancerRequestEvent`).